src/systems/filecoin_mining/storage_mining/storage_miner_actor.go

```diff
@@ -14,14 +14,46 @@ func (st *SectorTable_I) InactivePower() block.StoragePower {
 	return block.StoragePower((st.CommittedSectors_ + st.RecoveringSectors_ + st.FailingSectors_) * util.UVarint(st.SectorSize_))
 }
 
+func (cs *ChallengeStatus_I) OnNewChallenge(currEpoch block.ChainEpoch) ChallengeStatus {
+	cs.LastChallengeEpoch_ = currEpoch
+	return cs
+}
+
+// Call by either SubmitPoSt or OnMissedPoSt
+// TODO: verify this is correct and if we need to distinguish SubmitPoSt vs OnMissedPoSt
+func (cs *ChallengeStatus_I) OnChallengeResponse(currEpoch block.ChainEpoch) ChallengeStatus {
+	cs.LastResponseEpoch_ = currEpoch
+	return cs
+}
+
+func (cs *ChallengeStatus_I) isChallenged() bool {
+	// true (isChallenged) when LastChallengeEpoch is later than LastResponseEpoch
+	return cs.LastChallengeEpoch() > cs.LastResponseEpoch()
```

Comment:
small naming note, i'd expect from naming alone that `LastResponseEpoch` would only be set by SubitPost, but this needs to be set by the `onMissedPost()` flow as well. Maybe `LastChallengeEndEpoch` (sounds worse, but more clear?) or something like it